GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -408,10 +410,18 @@ public void OnFilePreDelete(string relativePath)
             this.backgroundFileSystemTaskRunner.Enqueue(FileSystemTask.OnFilePreDelete(relativePath));
         }
 
-        public void OnFolderCreated(string relativePath)
+        public bool OnFolderCreated(string relativePath)
         {
+            GitIndexProjection.PathProjectionState pathProjectionState = this.GitIndexProjection.GetPathProjectionState(relativePath);
+            if (pathProjectionState == GitIndexProjection.PathProjectionState.Excluded)
+            {
+                this.GitIndexProjection.TryAddIncludedFolder(relativePath);

```

Comment:
Do we care about success / failure here? When might this be ok to fail?

Reply:
Good point.  I will check the return code before returning `true`.  My one concern here would be that a user adds a folder that is excluded that would have millions of files and folders and if we fail adding it to the include it will now be in the modified paths and could cause performance issues for which they can't fix without a full dehydrate until we get the dehydrate folders finished.  But this shouldn't happen all that often so I am okay with this.